### PR TITLE
Support all hpa types in dangling-hpa

### PR DESCRIPTION
**Pull request by charlesoconor**
closes https://github.com/stackrox/kube-linter/issues/389

### CODE REVIEW

**Review by janisz — APPROVED** [review]

**Review by dhaus67 — APPROVED** [review]

**Comment by charlesoconor** [comment]
Thanks for the approvals! Is there anything else I need to do before this is merged? I don't have write access so need someone to do it for me.

**Pull request merged**
